core/pkg/sync/http/http_sync.go

```diff
@@ -192,3 +246,61 @@ func (hs *Sync) Fetch(ctx context.Context) (string, error) {
 	body, _, err := hs.fetchBody(ctx, false)
 	return body, err
 }
+
+func (hs *Sync) getClient() Client {
+	if hs.oauthCredential != nil {
+		delta := time.Since(hs.oauthCredential.lastUpdate).Seconds()
+		if delta <= float64(hs.oauthCredential.reloadDelaySeconds) && hs.client != nil {
```

Comment:
[question] Isn't this suppose to be handled automagically by the oauthCredentials - at least that would be my expectations, sorry i did not investigate further, but i assume you have already more insights than me :) Or what is the purpose for this reloadDelaySeconds?

Reply:
The OAuth implementation handles automatically the refresh of tokens when they expire. This logic is to reload the client and secret after `reloadDelaySeconds`. I get custom to this behavior in K8s since secrets are usually mounted via ConfigMaps or secret injection on the file system. This reload of the auth params is handy-dandy to support rotation of secrets w/o restarting the container and dealing with mounting delays.